api/client/swarm/init.go

```diff
@@ -36,7 +37,8 @@ func newInitCommand(dockerCli *client.DockerCli) *cobra.Command {
 	}
 
 	flags := cmd.Flags()
-	flags.Var(&opts.listenAddr, "listen-addr", "Listen address")
+	flags.Var(&opts.listenAddr, flagListenAddr, "Listen address")
+	flags.StringVar(&opts.listenInterface, flagListenInterface, "", "Listen on an interface's address")
```

Comment:
We already do the colon on --cluster-advertise, which accepts an ip or an interface.